src/app/test/business/dad_jokes_service_test.dart

```diff
@@ -0,0 +1,125 @@
+import 'dart:convert';
+
+import 'package:app/access/dad_jokes/dad_jokes_repository.dart';
+import 'package:app/access/dad_jokes/data/dad_joke_content_data.dart';
+import 'package:app/access/dad_jokes/favorite_dad_jokes_repository.dart';
+import 'package:app/business/dad_jokes/dad_joke.dart';
+import 'package:app/business/dad_jokes/dad_jokes_service.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../mocks/dad_jokes_data_mock.dart';
+import '../mocks/dad_jokes_list_mock.dart';
+import 'dad_jokes_service_test.mocks.dart';
+
+@GenerateNiceMocks(
+  [
+    MockSpec<DadJokesRepository>(),
+  ],
+)
+void main() {
+  late DadJokesRepository dadJokesRepository;
+  late FavoriteDadJokesRepository favoriteDadJokesRepository;
+  late DadJokesService SUT;
+
+  setUp(() {
+    TestWidgetsFlutterBinding.ensureInitialized();
+    SharedPreferences.setMockInitialValues({
+      "favoriteDadJokes":
+          getFavoriteDadJokesList().map((dadJoke) => jsonEncode(dadJoke)).toList()
+    });
+
+    provideDummy(mockedDadJokeResponse);
+    dadJokesRepository = MockDadJokesRepository();
+    when(dadJokesRepository.getDadJokes()).thenAnswer(
+      (_) async => mockedDadJokeResponse,
+    );
+
+    favoriteDadJokesRepository = FavoriteDadJokesRepository();
+    SUT = DadJokesService(dadJokesRepository, favoriteDadJokesRepository);
+  });
+
+  test('Get jokes', () async {
+    var result = await SUT.getDadJokes();
```

Comment:
Maybe it's not that mandatory, but normally test should be written with explicit Arrange, Assert & Assert sections. It gives structure to the tests